From 059fa599dd852fdcb508cc96e5cb3e0dc71689fb Mon Sep 17 00:00:00 2001
From: khanhluu <khanh.luu.xw@rvc.renesas.com>
Date: Tue, 10 Apr 2018 08:33:19 +0700
Subject: [PATCH 120/433] spi: sh-msiof: Fix MSIOF address for DMAC

Reference path:
- git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-backport.git
- commit: <a4e1adadc23c9fbbee8181f1d1b545b6b2dedea2>

    MSIOF Base Address H'E6xx can be accessed by CPU and DMAC.
    MSIOF Base Address H'E7xx for DMAC was removed from H/W manual.

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
Signed-off-by: khanhluu <khanh.luu.xw@rvc.renesas.com>
Signed-off-by: Quynh Nguyen <quynh.nguyen.xb@renesas.com>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 drivers/spi/spi-sh-msiof.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/spi/spi-sh-msiof.c b/drivers/spi/spi-sh-msiof.c
index 1470538..600fa07 100644
--- a/drivers/spi/spi-sh-msiof.c
+++ b/drivers/spi/spi-sh-msiof.c
@@ -1164,9 +1164,7 @@ static int sh_msiof_request_dma(struct sh_msiof_spi_priv *p)
 	}
 
 	/* The DMA engine uses the second register set, if present */
-	res = platform_get_resource(pdev, IORESOURCE_MEM, 1);
-	if (!res)
-		res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
+	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 
 	master = p->master;
 	master->dma_tx = sh_msiof_request_dma_chan(dev, DMA_MEM_TO_DEV,
-- 
2.7.4

